api_review/RadialGradientBrush/RadialGradientBrush.md

```diff
@@ -0,0 +1,244 @@
+
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is:
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+This example creates a radial gradient with six gradient stops and uses it to paint a Rectangle.
+
+![A rectangle filled with a radial gradient.](images/ColorRadialGradientBrush.png)
+
+XAML
+```XAML
+<Page
+  xmlns:media="using:Microsoft.UI.Xaml.Media">
+
+  <Rectangle Width="200" Height="200">
+      <Rectangle.Fill>
+          <media:RadialGradientBrush>
+              <GradientStop Color="Blue" Offset="0.0" />
+              <GradientStop Color="Yellow" Offset="0.2" />
+              <GradientStop Color="LimeGreen" Offset="0.4" />
+              <GradientStop Color="LightBlue" Offset="0.6" />
+              <GradientStop Color="Blue" Offset="0.8" />
+              <GradientStop Color="LightGray" Offset="1" />
+          </media:RadialGradientBrush>
+      </Rectangle.Fill>
+  </Rectangle>
+
+</Page>
+```
+
+This example creates a radial gradient with that uses Absolute mapping mode with custom values for `EllipseCenter`, `EllipseRadius` and `GradientOriginOffset`:
+
+![A rectangle filled with an offset radial gradient.](images/OffsetRadialGradientBrush.png)
+
+```XAML
+<Page
+  xmlns:media="using:Microsoft.UI.Xaml.Media">
+
+  <Rectangle Width="200" Height="200">
+      <Rectangle.Fill>
+          <media:RadialGradientBrush
+            MappingMode="Absolute"
+            EllipseCenter="50,50"
+            EllipseRadius="100,100"
+            GradientOriginOffset="50,0"
+            >
+              <GradientStop Color="Yellow" Offset="0.0" />
+              <GradientStop Color="Blue" Offset="1" />
+          </media:RadialGradientBrush>
+      </Rectangle.Fill>
+  </Rectangle>
+
+</Page>
+```
+
+# Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+## Gradient layout
+
+The gradient is drawn within an ellipse that is defined by the `EllipseCenter` and `EllipseRadius` properties. Colors for the gradient start at the center of the ellipse and end at the radius.
+
+The colors for the radial gradient are defined by color stops added to the `GradientStops` collection property. Each gradient stop specifies a color and an offset along the gradient.
+
+The gradient origin defaults to center and can be offset using the `GradientOriginOffset` property.
```

Comment:
This "OriginOffset" approach of offsetting from center is currently consistent with [CompositionRadialGradientBrush](https://docs.microsoft.com/uwp/api/windows.ui.composition.compositionradialgradientbrush), vs. WPF which instead had an "[Origin](https://docs.microsoft.com/dotnet/api/system.windows.media.radialgradientbrush.gradientorigin?view=netframework-4.8#System_Windows_Media_RadialGradientBrush_GradientOrigin)" property. 

Is it better to have closer WPF copy-pasteability or to be consistent with newer WinUI/WinRT APIs?

Reply:
I prefer being consistent within WinUI/WinRT itself. 

In the beginning, may new developers to WinUI (Desktop) will probably have a rather strong background in WPF. As the WinUI developer story will mature over the coming years though, I believe more and more developers will join who don't necessarily have that strong background. They will expect and should be provided with an in itself consistent platform for the best developer experience.

Reply:
Discussion:
WPF and default value is 0.5,0.5
Composition Origin is the middle and then offsets from that.
Both have 0,0 at the upper left

Seems odd for 0,0 not to be the "origin"

Summary:
In Comp you can't set the origin but you can offset the origin. (So it's an Origin _Offset_ property)
In WPF you set the origin but you can't offset it. (Origin property defaults to 0.5,0.5)

So what should the origin be, upper left like WPF or center like Comp?

Let's follow up on the reasoning behind the Comp choice of origin concept. Pending new information, let's align with WPF for consistency within the two Xaml APIs.